t/helper/test-quote.c

```diff
@@ -0,0 +1,30 @@
+#include "test-tool.h"
+#include "quote.h"
+#include "strbuf.h"
+#include "string.h"
+
+int cmd__quote_buf_pretty(int argc, const char **argv)
+{
+	struct strbuf buf_payload = STRBUF_INIT;
+	char nullString[2];
```

Comment:
This won't actually allocate a null string, unless my C is very rusty. Maybe try `char* nullString = 0` which should expose a panic in the `sq_quote_buf_pretty` where we dereference the pointer when testing for null or empty.